### DISCUSSION TITLE
GRPC integration with istio

### DISCUSSION THREAD

**Discussion by ghost** [istio/istio#48810] [Q&A]
Hi istio community
Hope you can help me with this.
I have two services, let's say service A and service B, so service A communicates with B using grpc everything works fine within a K8S cluster but my problem comes when I add istio on the cluster, for example if I kill pod of my service B my service A is unable to communicate with B now I checked the istio-proxy logs and receive a 200 UH no healthy upstream everytime my service A attempts a reconnection to service B after Pod B was restarted.
Do you have any idea what causes this?,
Is it a problem with the load balancer since I use service DNS to reach the GRPC service?.

**Comment by howardjohn** [comment]
UH means there is no healthy backends to connect to. Is there any healthy pods in Service B when it happens? You can check with `kubectl get endpoints` (k8s view) and `istioctl proxy-config endpoints` (istio view)

**Reply by ghost** [reply]
No due to design I only have one replica for my service B pod but this only happens when I add istio since without after the pods restart my service A groc client is able to connect again to the service B

I check with kubectl get ep and can see my grpc service endpoint there after it restarts the pod
